File: src/2015/Day_03/P1.rb
data = File.read("input.txt").split("")

houses = ["0:0"]
checked = 1

pos = [0, 0]

for token in data
    if token === "<"
        pos[0] -= 1
    elsif token === ">"
        pos[0] += 1
    elsif token === "^"
        pos[1] += 1
    else
        pos[1] -= 1
    end

    coord = "#{pos[0]}:#{pos[1]}"
    if houses.include? coord
        next
    else
        houses.append(coord)
        checked += 1
    end
end

puts "Result: #{checked}"

File: src/2024/Day_04/P1.ts
import * as path from 'path';
import * as fs from 'fs';
import { fileURLToPath } from 'url';

import { searchForWord } from '../../../utils/utils';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const data = fs.readFileSync(path.resolve(__dirname, `./input.txt`), `utf-8`);

const DIRS = [
    [-1, -1],
    [-1, 0],
    [-1, 1],
    [1, -1],
    [1, 0],
    [1, 1],
    [0, -1],
    [0, 1]
];

const main = async () => {
    const grid = data.split(`\n`).map(x => x.split(``));

    let ans = 0;
    for (let i = 0; i < grid.length; i++) {
        for (let j = 0; j < grid[0].length; j++) {
            // eslint-disable-next-line curly
            for (const dir of DIRS) {
                if (searchForWord(grid, `XMAS`, j, i, dir[0], dir[1])) ans++;
            }
        }
    }

    fs.writeFileSync(path.resolve(__dirname, `./output.txt`), String(ans));

    if (process.argv.length <= 3) console.log(`Result: ${ans}`);
    return ans;
};

if (process.argv.length <= 3) void main();
export default main;
